src/platform_impl/macos/appkit/cursor.rs

```diff
@@ -9,9 +9,19 @@ use objc2::runtime::Sel;
 use objc2::{extern_class, extern_methods, msg_send_id, mutability, sel, ClassType};
 
 use super::{NSBitmapImageRep, NSImage};
-use crate::cursor::CursorImage;
+use crate::cursor::{CursorImage, OnlyCursorImageBuilder};
+use crate::platform_impl::EventLoopWindowTarget;
 use crate::window::CursorIcon;
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct CustomCursor(pub(crate) Id<NSCursor>);
+
+impl CustomCursor {
+    pub fn build<T>(cursor: OnlyCursorImageBuilder, _: &EventLoopWindowTarget<T>) -> CustomCursor {
+        Self(NSCursor::from_image(&cursor.0))
+    }
+}
+
```

Comment:
I'd like to keep the `appkit` module free from `winit`-specific things like this, since I intend to fully replace the module with `icrate::AppKit`

Reply:
Where should I move it to, `window.rs`, `mod.rs`?

Reply:
I think I'd prefer it under a new file `macos/cursor.rs`, but eh, I'm going to do it myself in https://github.com/rust-windowing/winit/pull/2982, this is fine here for now.

Reply:
Too late, I've gone ahead and done it now :slightly_smiling_face: